nvflare/edge/aggregators/edge_dict_accumulator.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import Shareable
+from nvflare.app_common.abstract.aggregator import Aggregator
+from nvflare.edge.constants import MsgKey
+
+
+class EdgeDictAccumulator(Aggregator):
+    def __init__(self):
+        Aggregator.__init__(self)
+        self.weights = None
+        self.num_devices = 0
+
+    def _aggregate(self, weight_base, weight_to_add):
+        # aggregates the dict on corresponding keys
+        for key, sub_object in weight_base.items():
+            if isinstance(sub_object, dict):
+                sub_to_add = weight_to_add.get(key)
+                self._aggregate(sub_object, sub_to_add)
+            weight_base[key] = np.add(weight_base[key], weight_to_add[key])
+
+        return weight_base
```

Comment:
This return value is a ndarray now. Make sure that's expected type.

Reply:
Thanks! updated